[TASK]: Add a emailer and send confirmation email

sandipbhuyan opened the pull request:
Closes #142 

I used the feature of FOSUserBundler. I have added two extra parameters i.e mailer_encryption and  mailer_auth_mode for the encryption and authentication of mailing services.

The confirmation link is going through an emailer template which is in emailer.html.twig.

The email which the user will receive will be like:  
![screen shot 2018-02-24 at 9 31 19 pm](https://user-images.githubusercontent.com/18291428/36632233-199871a2-19aa-11e8-827c-7570645fafc8.png)

sandipbhuyan commented:
@agathver @imphil PR for issue #142

agathver commented on site/app/config/parameters.yml.dist:
```diff
@@ -7,6 +7,8 @@ parameters:
     database_user:     librecores
     database_password: ~
 
+    mailer_auth_mode: login #the type of authentication the mailer is checking eg: 'login'
```
I'm not sure if this is required, the plan is to go with AWS SES, which supports plain auth`.

agathver commented on site/app/Resources/views/emailer.html.twig:
```diff
@@ -0,0 +1,154 @@
+                </td>
+            </tr>
+        </table>
+
```
It's better (and required in some legislations) to add a mail footer: https://codepen.io/rodriguezcommaj/pen/RNPzwr

Please add LibreCores contact address and why they received this email something like "you have received this email as you or someone posing as you have signed-up at LibreCores"

sandipbhuyan replied:
I want the content for the mail footer. i have checked the librecores website but didn't get any address which i can use in the footer

agathver commented on site/app/config/config.yml:
```diff
@@ -160,9 +162,17 @@ fos_user:
+                    address:        'noreply@librecores.org'
+                    sender_name:    'LibreCores'
+                enabled:    true # change to true for required email confirmation
+                template:   'emailer.html.twig' #custom emailer template
```
Instead of assigning a custom template name, I would prefer you to override the default FOSUserBundle templates as in https://github.com/librecores/librecores-web/pull/193

Also add a plain-text version of the email for better support of mobile and command-line clients.

imphil replied:
You could use: "LibreCores is a project by The Free and Open Source Silicon Foundation C.i.C., 71-75 Shelton Street, London, WC2H 9JQ, United Kingdom"

sandipbhuyan commented:
@agathver sandy refers to the username

sandipbhuyan commented:
@agathver @imphil I have made some changes 
The email the users are getting will be like
![screen shot 2018-02-25 at 8 08 15 pm](https://user-images.githubusercontent.com/18291428/36642628-4f81d3a8-1a68-11e8-9b63-af791a3721ce.png)
And I have modified the README of sites directory.I have added details about salt email template.

agathver commented on site/app/Resources/FOSUserBundle/views/Registration/email.html.twig:
```diff
@@ -0,0 +1,175 @@
+                                <table width="500" border="0" cellspacing="0" cellpadding="0" align="center" class="responsive-table">
+                                    <tr>
+                                        <td align="center" valign="middle" style="font-size: 12px; line-height: 18px; font-family: Helvetica, Arial, sans-serif; color:#666666;">
+                                            <span class="appleFooter" style="color:#666666;"> 71-75 Shelton Street, London, WC2H 9JQ, United Kingdom</span><br><a style="color: #666666; text-decoration: none;">you are receiving this email because you (or someone posing as you) signed up at LibreCores</a>
```
~you~ -> You

agathver commented:
@imphil, In the current form, this looks good to me. We should go ahead and configure SES to make this run.

imphil commented:
Thanks @sandipbhuyan for this PR. I've slightly modified it and added the necessary deployment code. There are a couple more things to be done on my side to correctly configure Amazon SES in all its glory, but we should be able to send out mails soon! I've also opened a couple follow-up issues (#212 , #211) with things I noted when giving the changeset a try.

The final commit was https://github.com/librecores/librecores-web/commit/e8c0eccdeaab551925d422f3b4d77d42f307b200

sandipbhuyan commented:
Thanks @imphil for merging this PR. I will be working shortly on the relative issues (#212 , #210 , #211 )
